include/deal.II/fe/fe_update_flags.h

```diff
@@ -410,6 +438,37 @@ namespace internal
       std::vector<DerivativeForm<1,spacedim,dim> > inverse_jacobians;
 
       /**
+       * Array of the derivatives of the Jacobian matrices at the
+       * quadrature points, pushed forward to the real cell coordinates.
+       */
+      std::vector<Tensor<3,spacedim> > jacobian_pushed_forward_grads;
+
+      /**
+       * Array of the second derivatives of the Jacobian matrices at the
+       * quadrature points.
+       */
+      std::vector<DerivativeForm<3,dim,spacedim> > jacobian_hessians;
+
+      /**
+       * Array of the  second derivatives of the Jacobian matrices at the
+       * quadrature points, pushed forward to the real cell coordinates.
+       */
+      std::vector<Tensor<4,spacedim> > jacobian_pushed_forward_hessians;
```

Comment:
From the description, I can't tell what exactly the difference between `jacobian_hessians` and `jacobian_pushed_forward_hessians` is. Can you elaborate?

As a user of FEValues, do I actually care about both of these, or is only one of them of interest to me? If the latter, then the field that is not of interest should probably not be here as it is used only internally (likely by the finite element class) and should then be one of the `mutable` fields in `FE_Whatever::InternalData`.